packages/middleware-bucket-endpoint/src/configurations.ts

```diff
@@ -0,0 +1,39 @@
+export interface BucketEndpointInputConfig {
+  bucketEndpoint?: boolean;
+  forcePathStyle?: boolean;
+  preformedBucketEndpoint?: boolean;
+  useAccelerateEndpoint?: boolean;
+  useDualstackEndpoint?: boolean;
+}
+
+export interface BucketEndpointResolvedConfig {
+  bucketEndpoint: boolean;
+  forcePathStyle: boolean;
+  preformedBucketEndpoint: boolean;
+  useAccelerateEndpoint: boolean;
+  useDualstackEndpoint: boolean;
+}
+
+export function resolveBucketEndpointConfig<T>(
+  input: T & BucketEndpointInputConfig
+): T & BucketEndpointResolvedConfig {
+  const bucketEndpoint = input.bucketEndpoint ? input.bucketEndpoint : false;
+  const forcePathStyle = input.forcePathStyle ? input.forcePathStyle : false;
+  const preformedBucketEndpoint = input.preformedBucketEndpoint
+    ? input.preformedBucketEndpoint
+    : false;
+  const useAccelerateEndpoint = input.useAccelerateEndpoint
+    ? input.useAccelerateEndpoint
+    : false;
+  const useDualstackEndpoint = input.useDualstackEndpoint
+    ? input.useDualstackEndpoint
+    : false;
```

Comment:
All of these can be change to

```javascript
const {
    bucketEndpoint = false,
    forcePathStyle = false,
//others
} = input;
```

Reply:
Note that defaulting only applies if the left-hand side is `undefined` - not if it's otherwise falsey like the existing code. May or may not be what you want.

Reply:
fixed.